From f075893e9b0e241879998c0b12cf8af0ba7737da Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Mon, 23 Oct 2023 10:03:03 +0100
Subject: [PATCH] drivers: media: imx477: Disable the scaler

The horizontal scaler was enabled for the 2028x1520 and 2028x1080 modes,
with a scale factor of 1. It caused a single column of bad pixels on the
right edge of the image. Since scaling is not needed for these modes,
disable it entirely.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/media/i2c/imx477.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/media/i2c/imx477.c
+++ b/drivers/media/i2c/imx477.c
@@ -632,7 +632,7 @@ static const struct imx477_reg mode_2028
 	{0x9e9f, 0x00},
 	{0xa2a9, 0x60},
 	{0xa2b7, 0x00},
-	{0x0401, 0x01},
+	{0x0401, 0x00},
 	{0x0404, 0x00},
 	{0x0405, 0x20},
 	{0x0408, 0x00},
@@ -733,7 +733,7 @@ static const struct imx477_reg mode_2028
 	{0x9e9f, 0x00},
 	{0xa2a9, 0x60},
 	{0xa2b7, 0x00},
-	{0x0401, 0x01},
+	{0x0401, 0x00},
 	{0x0404, 0x00},
 	{0x0405, 0x20},
 	{0x0408, 0x00},
